src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/DefaultLSPDocumentMappingProvider.cs

```diff
@@ -21,23 +21,23 @@ internal class DefaultLSPDocumentMappingProvider : LSPDocumentMappingProvider
         private static readonly TextEdit[] EmptyEdits = Array.Empty<TextEdit>();
 
         private readonly LSPRequestInvoker _requestInvoker;
-        private readonly LSPDocumentManager _documentManager;
+        private readonly Lazy<LSPDocumentManager> _lazyDocumentManager;
 
         [ImportingConstructor]
-        public DefaultLSPDocumentMappingProvider(LSPRequestInvoker requestInvoker, LSPDocumentManager documentManager)
+        public DefaultLSPDocumentMappingProvider(LSPRequestInvoker requestInvoker, Lazy<LSPDocumentManager> lazyDocumentManager)
```

Comment:
Was there a reason why you made `LSPDocumentManager` lazy here instead of making `LSPMappingProvider` lazy in `CSharpVirtualDocumentPublisher`? If not, it might be good to not involve this file given all other changes are happening in `CSharpVirtualDocumentPublisher`. In any case, It'd be good to have a comment indicating why we are making it lazy (whichever one you decide to make lazy)

Reply:
https://github.com/dotnet/aspnetcore-tooling/pull/2041#discussion_r441113371

Adding a comment :)